### ISSUE TITLE
CollectionView#destroyChildren should not call checkEmpty

### ISSUE THREAD

**Issue by cmaher** [marionettejs/backbone.marionette#2313]
```js
  destroyChildren: function() {
    var childViews = this.children.map(_.identity);
    this.children.each(this.removeChildView, this);
    this.checkEmpty();  // this is a problem
    return childViews;
  },
```
`checkEmpty` will show an emptyView if the collection is empty. `destroy` calls `destroyChildren`, which removes all children, and will then add one back (if the collection is empty). This is a leak.

Other methods calling `destroyChildren`:

* `destroyEmptyView` -- probably shouldn't be immediately showing another empty view
* `_renderChildren` -- manually calls `showEmptyView` when appropriate

Labeling as 'minor' because `destroyChildren` isn't publicly exposed in the docs.

**Comment by rhubarbselleven** [comment]
hey @cmaher I did some work around this a while back:

https://github.com/marionettejs/backbone.marionette/pull/1663
https://github.com/marionettejs/backbone.marionette/issues/1662

Where I got to with it was that I could not prove via unit test that the empty view was actually created.

**Comment by ahumphreys87** [comment]
Taking this further... should the `emptyView` even be added to `children`? I guess technically it is a 'child' but I always think of `children` as the views for the rendered collection.

Taking `emptyView` out of the `children` enumerable would us manage this better I think

**Comment by cmaher** [comment]
I was thinking of that too @ahumphreys87, but that's definitely a breaking change.  I think it makes sense when you consider that the purpose of `babysitter` is to manage subviews, and an `emptyView` is just another subview.

**Comment by ahumphreys87** [comment]
I have 2 ideas for a temporary fix:

1. Create a `_destroyAll` that is called from `destroy` instead of `destroyChildren`. This will not `checkEmpty`.
2. Pass an option to `destroyChildren` that will disable the `checkEmpty` call if set.

Not sold on either of them to be honest. Option 1 we would end up duplicating most of `destroyChildren`, option 2 just feels too hacky. Any other ideas @cmaher?

**Comment by cmaher** [comment]
The only thing I think is breaking is not adding the `emptyView` to `children`.  I'm pretty sure that calling `checkEmpty` is just a bug.

**Comment by ahumphreys87** [comment]
Im guessing it's there so that users can call `destroyChildren` to empty their view rather than resetting the collection. Its undocumented but I bet there are some people that will do this.

**Comment by ahumphreys87** [comment]
More thoughts: `destroyChildren` is useful in `CompositeView` to allow users to only clear children and show the empty view in its place. Id say this would definitely be a breaking change.

**Comment by cmaher** [comment]
I think you raise a good point there @ahumphreys87. I'm leaning toward your option 2, just for purposes of not duplicating code.

**Comment by cmaher** [comment]
resolved by #2384
